base/geo_object_id.hpp

```diff
@@ -0,0 +1,99 @@
+#pragma once
+
+#include <cstdint>
+#include <functional>
+#include <string>
+
+namespace base
+{
+// GeoObjectId is used to pack the source of a geographical object together with its
+// id within this source (that is, its serial number) into a single 64-bit number.
+//
+// The bit encoding is as follows:
+//   0sss ssss RRRR RRRR xxxx xxxx xxxx xxxx xxxx xxxx xxxx xxxx xxxx xxxx xxxx xxxx
+//   R - reserved bits
+//   s - bits for object source
+//   x - bits for serial number (object id within its source)
+//
+// The highest byte encodes one of (2^7 - 1) = 127 possible object sources.
+// Another byte is reserved and the last 6 bytes leave us with 2^48 possible values that can be
+// used for ids within a source.
+// Typically, the reserved byte will be zero but it may be used in future if the format changes.
+// At the time of writing, OSM has approximately 2^32 different nodes with ids starting
```

Comment:
Probably it's better to write the date you wrote about.

Reply:
I thought the commit date was enough.
done

Reply:
Thanks